drivers/soundwire/intel.c

```diff
@@ -1693,7 +1695,7 @@ static int __maybe_unused intel_suspend_runtime(struct device *dev)
 			return ret;
 		}
 
-		intel_shim_wake(sdw, true);
+		intel_shim_wake(sdw, wake_enable);
```

Comment:
this seems good to me, let's do this.

Minor edits to the commit message

dev->power.runtime_error will be set to the return value of the runtime
suspend callback function, and runtime resume function will return
-EINVAL if dev->power.runtime_error is not 0.

Somehow the codec rarely doesn't return an ACK to the clock prepare command.
If we stop the runtime suspend process and return error, we will not
be able to resume again. Likewise, if the codec lost sync and did not rejoin, the resume operation will also fail. As a result, the SoundWire bus can not be used
anymore.

This patch suggests to finish the runtime suspend process even if we
fail to stop sdw bus clock.  In the case where we do a hardware reset, the codecs will be reconfigured completely.
In the case where we use the regular clock stop, the codecs keep their state and worst case will fall off the bus and reattach.


The only drawback is that the power consumption may be higher and device-initiated interrupts may be lost, but at least audio function can still work after resume

Reply:
@plbossart Welcome back. Updated.